### PR TITLE
Add new `Style/RedundantDoubleSplatBrackets` cop

### PR DESCRIPTION
**Pull request by koic**
This cop checks for redundant uses of double splat hash braces.

```ruby
# bad
do_something(**{foo: bar, baz: qux})

# good
do_something(foo: bar, baz: qux)
```

Both are keyword arguments after all.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/redundant_double_splat_hash_braces.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      #   # bad
+      #   do_something(**{foo: bar, baz: qux})
+      #
+      #   # good
+      #   do_something(foo: bar, baz: qux)
+      #
+      class RedundantDoubleSplatHashBraces < Base
+        extend AutoCorrector
+
+        MSG = 'Remove the redundant double splat hash braces.'
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_double_splat_hash_braces.rb:19]: "double splat preceding the hash literal" sounds better to me.

Comment (koic) [reply lib/rubocop/cop/style/redundant_double_splat_hash_braces.rb:19]: Actually, braces are redundant as well as double splat.

```ruby
# bad
do_something(**{foo: bar, baz: qux})

# good
do_something(foo: bar, baz: qux)
```

For example, what about the following message?

```
Remove redundant double splat and braces, use keyword arguments directly.
```

Comment (bbatsov) [reply lib/rubocop/cop/style/redundant_double_splat_hash_braces.rb:19]: Fine by me.

Comment (koic) [reply lib/rubocop/cop/style/redundant_double_splat_hash_braces.rb]: I've updated it. Thank you!

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**